src/components/home/balance-card.tsx

```diff
@@ -22,7 +22,9 @@ const BalanceCard = () => {
   const userProfile = useAtomValue(userProfileAtom);
   const setError = useSetAtom(errorAtom);
   const [loading, setLoading] = useState(false);
+  const [fetchComplete, setFetchComplete] = useState(false);
```

Comment:
As far as I can tell this is unnecessary and can be removed. If you think it's needed can you describe how I can replicate the issue that means it's necessary?